src/targets/gpu/kernels/include/migraphx/kernels/roialign.hpp

```diff
@@ -92,11 +93,14 @@ MIGRAPHX_DEVICE_CONSTEXPR typename Iterator::value_type bilinear_interpolate(
                                 high[0] * dims[1] + low[1],
                                 high[0] * dims[1] + high[1]};
 
-    float ly                                   = xy[0] - low[0];
-    float lx                                   = xy[1] - low[1];
-    float hy                                   = 1.0f - ly;
-    float hx                                   = 1.0f - lx;
-    array<typename Iterator::value_type, 4> ws = {hy * hx, hy * lx, ly * hx, ly * lx};
+    float ly              = xy[0] - low[0];
+    float lx              = xy[1] - low[1];
+    float hy              = 1.0f - ly;
+    float hx              = 1.0f - lx;
+    array<ret_type, 4> ws = {static_cast<ret_type>(hy * hx),
```

Comment:
We should probably just make the array a `float`, and then cast back to the original type at the end.

Reply:
Done.